### PR TITLE
- IP Enrichment - Generic v2 + Test Playbook, Entity Enrichment - Generic v2, Entity Enrichment - Phishing v2

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
In-Progress:
1. Fix `Process Email - Generic` bug, and release as v2
2. Duplicate test for phishing v2

## Related Issues
Related to: https://github.com/demisto/etc/issues/14480

## Description
Added a new playbook for IP enrichment, named `IP Enrichment - Generic v2`. The following changes were made:
- `IP Enrichment - Generic v2` is split into 2 separated playbooks. One for internal IPs, the other for external ones
- Added `IP Enrichment - External - Generic` - a playbook for enrichment of external IPs only.
- Added `IP Enrichment - Internal - Generic` - a playbook for enrichment of internalIPs only.
- Added enrichment of IPs via Threat Crowd
- New `Entity Enrichment - Phishing` playbook, which is a slim version of the generic one, and was made to avoid enrichment of irrelevant indicators in phishing investigations (for example, only enriches external IPs, and does not enrich endpoints)
- **Uniq** transformer applied in multiple locations to prevent enrichment duplication
- Auto-extract set to correct types in playbook tasks, to improve performance and indicator extraction, enrichment and reputation
- Added "ignore case" where relevant to prevent mistakes with enrichment and with the test
- Added a new test playbook for `IP Enrichment`, which does not use VirusTotal, but Threat Crowd - this should prevent quota limitations that we have with VT
- Added a new test script to be used by the new `IP Enrichment` test playbook
- Changed configuration of `IsIPInRanges` script, so that the script is not mistakenly shown as a condition-script.
- Added a flag for `resolveIP` to `Entity Enrichment - Generic` and to `Entity Enrichment - Phishing`, which will be passed down to `IP Enrichment - Generic` and its sub-playbooks (and in the future, also to other enrichment sub-playbooks).
- Added a flag for `GetReputation` to `Entity Enrichment - Generic` and to `Entity Enrichment - Phishing`, which will be passed down to `IP Enrichment - Generic` and its sub-playbooks (and in the future, also to other enrichment sub-playbooks).
- Small fix in `test_integration.py`, done by @anara123 to prevent sync VS async problems

## Required version of Demisto
4.1.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Code Review

### CODE REVIEW

**Comment by Itay4** [comment]
@idovandijk what is the status here?

**Comment by idovandijk** [comment]
> @idovandijk what is the status here?

Same as the other PR, only this one is going to be merged last since it contains Entity Enrichment (one of the parent playbooks)

File: Tests/conf.json [hunk 1]
```diff
@@ -1324,7 +1354,7 @@
         },
         {
             "integrations": "ANYRUN",
-            "playbookID": "Detonate File From URL - ANYRUN - Test"
+            "playbookID": "Palo Alto Networks Cortex Test"
```
Comment (Itay4) [inline Tests/conf.json]: why did you change this??

Comment (idovandijk) [reply Tests/conf.json]: Mistake, I was having lots of conflicts in my PR and must have thought this was accident that I made.
